### PR TITLE
Fixes issues with missing data crashing the client.

### LINKED ISSUE #643: Crash on unit repair.
Its has been reported that vanilltd crashes when attempting to repair a unit on the service depo if the player has money and the unit is damaged.

### PR DESCRIPTION
**Pull request by OmniBlade**
Fixes issues such as repair voice missing in default TD data.
Fixes #643

Supercedes #645 and #504

### CODE REVIEW

File: common/rawfile.cpp [hunk 1]
```diff
@@ -93,10 +93,7 @@
  *=============================================================================================*/
 void RawFileClass::Error(int error, int canretry, char const* filename)
 {
-#ifndef _WIN32
     perror(filename);
```
Comment (giulianobelinassi) [inline common/rawfile.cpp]: On Windows, I would call MessageBox once the user won't have a command prompt attached to the executable.

If SDL2 build is enabled, we could even use SDL's messagebox.

**Review by giulianobelinassi — CHANGES_REQUESTED** [review]

File: common/rawfile.cpp (hunk 1 above)
Comment (OmniBlade) [reply common/rawfile.cpp]: I'd rather not prompt the user at all if they aren't deliberately trying to debug, though I could put this through the normal logging interface instead. I don't think the appropriate fix to the missing repair voice for example is for a box to pop up telling the user its missing when they probably don't care.

Comment (giulianobelinassi) [reply common/rawfile.cpp]: Using the logging interface may be more appropriate. (calls to DBG_LOG or CCDebugString)

Comment (OmniBlade) [reply common/rawfile.cpp]: I've updated the logging and also made sure that CCFileClass::Error actually passes the call down the chain.

**Comment by giulianobelinassi** [comment]
doesn't DBG_ERROR crashes the program?

**Comment by OmniBlade** [comment]
No, only DBG_FATAL is actually fatal.

**Review by giulianobelinassi — APPROVED** [review]

**Pull request merged**